kernel/unwind.c

```diff
@@ -0,0 +1,582 @@
+#include "unwind.h"
+
+#include "com.h"
+#include "util.h"
+
+extern const char _eh_frame_start[];
+extern const char _eh_frame_end[];
+
+/* DWARF data structs */
```

Comment:
Design note: the DWARF stuff could go in another file, but would it be useful in other places in the kernel..? I don't know. Note that there *is* already an `elf.h` file, so this could perhaps go in the corresponding `elf.c` file ..?

Also I don't know anything about DWARF other than that there are lots of conflicting versions.  What version does this support?